memory/memory/client.py

```diff
@@ -0,0 +1,79 @@
+from asyncinit import asyncinit
+import aiohttp
+import concurrent
+
+from hailtop.auth import service_auth_headers
+from hailtop.config import get_deploy_config
+from hailtop.google_storage import GCS
+from hailtop.tls import get_context_specific_ssl_client_session
+from hailtop.utils import async_to_blocking, request_retry_transient_errors
+
+@asyncinit
+class AsyncMemoryClient:
+    async def __init__(self, billing_project, fs=None, deploy_config=None, session=None,
+                       headers=None, _token=None):
+        self.billing_project = billing_project
+
+        if not deploy_config:
+            deploy_config = get_deploy_config()
+
+        self.url = deploy_config.base_url('memory')
+
+        if session is None:
+            session = get_context_specific_ssl_client_session(
+                raise_for_status=True,
+                timeout=aiohttp.ClientTimeout(total=60))
+        self._session = session
+
+        if fs is None:
+            fs = GCS(blocking_pool=concurrent.futures.ThreadPoolExecutor(), project=billing_project)
```

Comment:
I'm a little confused by this.  `billing_project` normally refers to a Hail billing project, not a Google project.